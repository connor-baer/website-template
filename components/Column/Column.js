import styled from '@emotion/styled';
import { css } from '@emotion/core';

const baseStyles = ({ theme, index }) => css`
  grid-area: ${`column-${index}`};
  align-self: center;
  display: flex;
  flex-direction: column;

  ${theme.mq.kilo} {
    > *:first-child {
      padding-top: 0;
      margin-top: 0;
    }

    > *:last-child {
      padding-bottom: 0;
      margin-bottom: 0;
    }
  }
`;

const Column = styled('div')(baseStyles);

export default Column;
