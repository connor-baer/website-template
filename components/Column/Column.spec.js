import React from 'react';

import Column from './Column';

describe('Column', () => {
  function renderColumn(fn, props = {}) {
    return fn(<Column {...props} />);
  }

  it('should render with default styles', () => {
    const children = 'label';
    const { container } = renderColumn(render, { children, index: 1 });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should meet basic accessibility guidelines', async () => {
    const children = 'label';
    const html = renderColumn(renderToHtml, { children, index: 1 });
    const result = await axe(html);
    expect(result).toHaveNoViolations();
  });
});
