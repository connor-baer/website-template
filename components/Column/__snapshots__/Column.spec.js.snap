// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column should render with default styles 1`] = `
.circuit-0 {
  grid-area: column-1;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media (min-width:600px) {
  .circuit-0 > *:first-child {
    padding-top: 0;
    margin-top: 0;
  }

  .circuit-0 > *:last-child {
    padding-bottom: 0;
    margin-bottom: 0;
  }
}

<div
  class="circuit-0 circuit-1"
>
  label
</div>
`;
